### DISCUSSION TITLE
How to restrict import a file?

### DISCUSSION THREAD

**Discussion by IRediTOTO** [eslint/eslint#16589] [Q&A] [answered]
Hi, I have a folder like this
components
    comA
        child1.ts
        child2.ts
        index.ts
   comB
   
How can only allowed child1, child2 in index.ts of folder comA?
child1.ts can't import to comB

**Comment by mdjermanovic** [comment, marked as answer]
You can use these rules:

https://eslint.org/docs/latest/rules/no-restricted-imports

https://github.com/import-js/eslint-plugin-import/blob/HEAD/docs/rules/no-restricted-paths.md

For restricting project files, `import/no-restricted-paths` is probably better.
